### PR TITLE
zsh,git: update caveats

### PR DESCRIPTION
**Pull request by bxsx**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

# Abstract

This PR adds caveats to `zsh` and `git` formulas.

# Background

It's a long story that confuse many users about Git completions in Zsh, see:
- #59062
- #52324
- #33157
- #32081
- #13700

While the general Homebrew policy is to install everything from upstream, it's not clear why the Git upstream takes precedence over the Zsh upstream in this case, especially when considering the statement provided by Git regarding their own completions:

https://github.com/git/git/blob/f9972720e9a405e4f6924a7cde0ed5880687f4d0/contrib/README#L3-L8

> Although these pieces are available as part of the official git
source tree, they are in somewhat different status.  The
intention is to keep interesting tools around git here, maybe
even experimental ones, to give users an easier access to them,
and to give tools wider exposure, so that they can be improved
faster.

https://github.com/git/git/blob/f9972720e9a405e4f6924a7cde0ed5880687f4d0/contrib/README#L26-L30

> I expect that things that start their life in the contrib/ area
to graduate out of contrib/ once they mature, either by becoming
projects on their own, or moving to the toplevel directory.  On
the other hand, I expect I'll be proposing removal of disused
and inactive ones from time to time.

# Proposal

This PR doesn't alter the current behavior of Homebrew, nor does it favor one completion over the other. However, it does help mitigate confusion that users might experience, particularly when the default completion (e.g. bundled with macOS) offers completions from the Zsh upstream that are overwritten by the `git` formula.

### CODE REVIEW

**Comment by SMillerDev** [comment]
I think these are way too wordy. And I guess aren't really Homebrew specific.

**Comment by bxsx** [comment]
> I think these are way too wordy. 

Hmm I've seen some noise caveats here and there, however, I'm open to suggestions or rewording! :)

> And I guess aren't really Homebrew specific.

I think it's quite Homebrew-specific. For example, macOS Ventura uses completions from Zsh, but I don't want to argue which upstream should take precedence since it's clear there is no will to change the current behavior. All I want is to mitigate the common confusion with some tips for users :)

```
$ ls ${^fpath}/_git(N)
/usr/zsh/5.9/share/zsh/functions/_git

$ /bin/ls -l /usr/share/zsh/5.9/functions/_git
-rw-r--r--  1 root  wheel  388215 Aug  2 11:44 /usr/share/zsh/5.9/functions/_git

$ /bin/ls -l /opt/homebrew/Cellar/zsh/5.9/share/zsh/functions/_git
-rw-r--r--  1 bart  admin  388215 May 14  2022 /opt/homebrew/Cellar/zsh/5.9/share/zsh/functions/_git

$ /bin/ls -l /opt/homebrew/Cellar/git/2.41.0_2/share/zsh/site-functions/_git
-rw-r--r--  1 bart  admin  7133 Jun  1 09:03 /opt/homebrew/Cellar/git/2.41.0_2/share/zsh/site-functions/_git
```

**Comment by SMillerDev** [comment]
> I think it's quite Homebrew-specific. For example, macOS Ventura uses completions from Zsh

But don't Arch, Debian and RHEL have the exact same issues?

**Comment by SMillerDev** [comment]
> BTW. Just checked on Debian. This is the same behavior as on macOS and is in line with the Git statement quoted above.
So, Debian only installs the completions from the Zsh package. Installing the git package does not install completions.

Did you check how Debian handles it or how Homebrew handles it on Debian? Because my point was to check if this is a Homebrew specific situation.

**Comment by fxcoudert** [comment]
Caveats are for things that are very specific to Homebrew. Documentation about choices of completions and paths are rather generic, so I agree with @SMillerDev that we probably do not want that.

**Comment by bxsx** [comment]
I believe this is very specific to Homebrew, so caveats should be suitable here. As mentioned in my previous comments, other distributions follow the official Git guidance, therefore, the Git package **does not include** Zsh completions at all.

Personally, I don't understand why Homebrew decided to go against the grain. Is there any rationale for this decision?@SMillerDev @fxcoudert @MikeMcQuaid ?

> @SMillerDev Did you check how Debian handles it or how Homebrew handles it on Debian? Because my point was to check if this is a Homebrew specific situation.

I provided the output of installing these packages on Debian (and macOS). If you are asking for details, yes, e.g. Debian simply does not touch Zsh completion at all. In the package rules you can only find references to Bash completion:
https://sources.debian.org/src/git/1%3A2.1.4-2.1%2Bdeb8u6/debian/rules/#L140-L144

Of course, Debian does install Git completion provided by Zsh: https://sources.debian.org/src/zsh/5.0.7-5/debian/rules/#L34

PS. I don't mind closing this PR. I would prefer to change this Homebrew-specific behavior and follow official principles provided by Git. I'll be happy to open PR with the required changes if you are willing to accept it.

**Comment by MikeMcQuaid** [comment]
> Personally, I don't understand why Homebrew decided to go against the grain. Is there any rationale for this decision?@SMillerDev @fxcoudert @MikeMcQuaid ?

That it's more useful to include them than not and that maintainers (including myself) use them and find them useful.
